Assets/Global/Scripts/WaveSpawnArea.cs

```diff
@@ -0,0 +1,63 @@
+using UnityEngine;
+using System.Collections.Generic;
+using Random = UnityEngine.Random;
+
+public class WaveSpawnArea : MonoBehaviour
+{
+    [Header("Spawner Settings")]
+    [Tooltip("Prefab of the enemy to spawn")]
+    [SerializeField] public GameObject enemyPrefab;
+
+    [SerializeField] public int maxSpawnedEnemies;
+
+    [Header("Spawn Area Settings")]
+    [SerializeField] public Transform spawnArea;
+
+    private List<GameObject> activeEnemies = new List<GameObject>();
+
+    public bool center = false;
+
+    public bool waveDone = false;
+
+    private void Update()
```

Comment:
Would it be more efficient to subscribe to an enemy kill event and do a check once? I am not sure how many enemies we will have but looping over them in an update function seems potentially expensive

Reply:
I don't think this is a code that we will have forever. this can be changed when the enemy is calling his own death event. but this is not for every enemy just for the enemies in the current wave which will not be that many i think. I dont think we will have 200 enemies in 1 wave in a small room.

either way you want an update somehow to check if the wave is done to spawn the next wave.